tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_snapshot_access.py

```diff
@@ -0,0 +1,338 @@
+
+        log.info(f"Test {tc} passed")
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Cleaning Up")
+        log.info("Remove snapshots")
+        for subvol in subvol_list:
+            if snap_list:
+                for snap_name in snap_list:
+                    snapshot = {
+                        "vol_name": fs_name,
+                        "subvol_name": subvol,
+                        "snap_name": snap_name,
+                        "group_name": "subvolume_group1",
+                    }
+                    fs_util.remove_snapshot(client1, **snapshot)
+
+        log.info("Unmount NFS export")
+        for mount in nfs_mount_list:
+            client1.exec_command(sudo=True, cmd=f"umount -l {mount}", check_ec=False)
+        log.info("Removing the Export")
+        for nfs_export_name in export_name_list:
+            client1.exec_command(
+                sudo=True,
+                cmd=f"ceph nfs export delete {nfs_name} {nfs_export_name}",
+                check_ec=False,
+            )
+        log.info("Removing NFS Cluster")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph nfs cluster rm {nfs_name}",
+            check_ec=False,
+        )
+        log.info("Removing Subvolumes and Subvolumegroup")
+        for subvol in subvol_list:
+            subvolume = {
+                "vol_name": fs_name,
+                "subvol_name": subvol,
+                "group_name": "subvolume_group1",
+            }
+            fs_util.remove_subvolume(client1, **subvolume)
+        fs_util.remove_subvolumegroup(client1, **subvolumegroup)
+
+
+def validate_snap_nfs(client, mount_list, snap_list):
+    for mount in mount_list:
+        cmd_out, cmd_rc = client.exec_command(sudo=True, cmd=f"ls {mount}/.snap")
+        log.info(cmd_out)
+        for snap_name in snap_list:
+            if snap_name not in cmd_out:
+                cmd_list = [
+                    "ceph orch ps --daemon_type=nfs",
+                    "ceph nfs cluster info cephfs-nfs",
+                ]
+                for cmd in cmd_list:
+                    client.exec_command(sudo=True, cmd=cmd)
+                log.error(f"Failed to view snapshot {snap_name} in NFS mount {mount}")
+
+
+def create_snapshot_wrapper(client1, fs_util, subvol_list, snap_list):
+    with parallel() as p:
+        for subvol in subvol_list:
+            p.spawn(create_snapshot, client1, fs_util, subvol, snap_list)
+
+
+def create_snapshot(client1, fs_util, subvol, snap_list):
+    for snap_name in snap_list:
+        snapshot = {
+            "vol_name": "cephfs",
+            "subvol_name": subvol,
+            "snap_name": snap_name,
+            "group_name": "subvolume_group1",
+        }
+        fs_util.create_snapshot(client1, **snapshot)
+        time.sleep(20)
+
+
+def write_io_wrapper(client1, nfs_mount_list, dir_name, run_time):
```

Comment:
This function also can be used by others can you move this to cephfs utils V1.

Reply:
as this wrapper would call write_io of local module  which is preferred, i cant move this to fs_utils, but i can add a write_io_wrapper into fsutils for furture use.